src/logic/calculate.js

```diff
@@ -0,0 +1,15 @@
+import operate from './operate';
+
+const Calculate = ({ total, next, operator }, buttonName) => {
+  switch (buttonName) {
+    case '+/-':
+      return { total: total * -1, next: next * -1, operator };
+    case '%':
+      return operate(total, next, buttonName);
+    case 'AC':
+      return { total: 0, next: 0, operator };
+    default:
+      return operate(total, next, buttonName);
```

Comment:
### Heads-Up:
**we need to add more cases;** 
example;
 - if the `buttonName`  = `.` we should return `0.`
 - case `/`, `x`, and `=`